lib/StripeResource.js

```diff
@@ -282,15 +330,57 @@ StripeResource.prototype = {
         path: path,
       });
 
+      var requestRetries = numRetries || 0;
+
       req._requestEvent = requestEvent;
 
       req._requestStart = Date.now();
 
       self._stripe._emitter.emit('request', requestEvent);
 
       req.setTimeout(timeout, self._timeoutHandler(timeout, req, callback));
-      req.on('response', self._responseHandler(req, callback));
-      req.on('error', self._errorHandler(req, callback));
+
+      req.on('response', function(res) {
+        if (self._shouldRetry(res, requestRetries)) {
+          requestRetries += 1;
+
+          // if it's a POST request, add an idempotency key
+          if (method === 'POST') {
+            headers = addIdempotentKey(headers);
```

Comment:
Okay this goes with my other comment, if you add the idempotency key on retry, you cause 2 requests to be sent.

Reply:
I misunderstood how idempotency works. Should be fixed now though!